inference-engine/src/readers/onnx_reader/onnx_model_validator.cpp

```diff
@@ -0,0 +1,172 @@
+        bytes.reserve(4);
+
+        char key_component = 0;
+        model.get(key_component);
+
+        while (key_component & 0x80) {
+            // drop the most significant bit
+            const char component = key_component & ~0x80;
+            bytes.push_back(component);
+            model.get(key_component);
+        }
+        // add the last byte - the one with MSB off
+        bytes.push_back(key_component);
+
+        return varint_bytes_to_number(bytes);
+    }
+
+    PbKey decode_key(const char key) {
+        // 3 least significant bits
+        const char wire_type = key & 3;
+        // remaining bits
+        const char field_number = key >> 3;
+        return {field_number, wire_type};
+    }
+
+    ONNXField decode_next_field(std::istream& model) {
+        char key = 0;
+        model.get(key);
+
+        const auto decoded_key = decode_key(key);
+
+        if (!is_correct_onnx_field(decoded_key.first)) {
+            throw std::runtime_error{"Incorrect field detected in the processed model"};
+        }
+
+        const auto onnx_field = static_cast<Field>(decoded_key.first);
+
+        switch (decoded_key.second) {
+            case VARINT: {
+                // the decoded varint is the payload in this case but its value doesnt matter
+                // in the fast check process so we just discard it
+                decode_varint(model);
+                return {onnx_field, 0};
+            }
+            case LENGTH_DELIMITED:
+                // the varint following the key determines the payload length
+                return {onnx_field, decode_varint(model)};
+            case BITS_64:
+                return {onnx_field, 8};
+            case BITS_32:
+                return {onnx_field, 4};
+            case START_GROUP:
+            case END_GROUP:
+                throw std::runtime_error{"StartGroup and EndGroup are not used in ONNX models"};
+            default:
+                throw std::runtime_error{"Unknown WireType encountered in the model"};
+        }
+    }
+
+    inline void skip_payload(std::istream& model, uint32_t payload_size) {
+        model.seekg(payload_size, std::ios::cur);
+    }
+} // namespace onnx
+} // namespace detail
+
+namespace InferenceEngine {
+    bool is_valid_model(std::istream& model, onnx_format) {
+        // the model should at least contain the ir_version, graph and opset_import fields
+        const unsigned int EXPECTED_FIELDS_FOUND = 3u;
+        unsigned int valid_fields_found = 0u;
+        try {
+            while (!model.eof() && valid_fields_found < EXPECTED_FIELDS_FOUND) {
+                const auto field = detail::onnx::decode_next_field(model);
+
+                ++valid_fields_found;
+
+                if (field.second > 0) {
+                    detail::onnx::skip_payload(model, field.second);
+                }
+            }
+
+            return valid_fields_found == EXPECTED_FIELDS_FOUND;
+        } catch (const std::exception& e) {
```

Comment:
I think we can use catch for any exceptions:
```suggestion
        } catch (...) {
```